auto_nag/topcrash.py

```diff
@@ -0,0 +1,97 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from datetime import datetime
+from typing import Dict, Optional, Union
+
+import libmozdata.socorro as socorro
+from libmozdata import utils as lmdutils
+
+
+class Topcrash:
+    def __init__(
+        self,
+        minimum_crashes: Optional[int] = 0,
+        minimum_startup_crashes: Optional[int] = 0,
+    ) -> None:
+        """Constructor
+
+        Args:
+            minimum_crashes: the minimum number of crashes to consider a
+                signature in the top crashes.
+            minimum_startup_crashes: the minimum number of crashes to consider a
+                signature as a startup crash.
+        """
+        self.minimum_crashes = minimum_crashes
+        self.minimum_startup_crashes = minimum_startup_crashes
+
+    def get_signatures(
+        self,
+        date: Union[str, datetime],
+        duration: Optional[int] = 11,
+        tc_limit: Optional[int] = 50,
+    ) -> Dict[str, bool]:
+        """Fetch the top crashes from socorro.
+
+        Top crashes will be queried twice for each release channel, one that
+        targets startup crashes and another that targets all crashes.
+
+        Args:
+            date: the final date
+            duration: the number of days to retrieve the crash data
+            tc_limit: the number of topcrashes to load on each query
+
+        Returns:
+            A dictionary where the keys are crash signatures, and the values are
+            booleans that indicate whether a crash is a startup crash or not.
+        """
+
+        start_date = lmdutils.get_date(date, duration - 1)
+        end_date = lmdutils.get_date(date)
+        date_range = socorro.SuperSearch.get_search_date(start_date, end_date)
+
+        params_combinations = [
```

Comment:
All the crashes you'll got here will be mostly the ones under Windows especially on release.
So probably having more granularity could be a good idea: but in a follow-up PR.

Reply:
Good point! It is simple to implement as part of this PR. We could have separate queries for each platform (i.e., Win, Mac, and Lin). This will generate 3X queries, which means maybe we should reduce the number of signatures (`tc_limit`) in each query.

Reply:
Improved in 87cf2f2224edd87e96f9244a7de0e52a2f779e70 to follow [the top crash identification criteria](https://wiki.mozilla.org/CrashKill/Topcrash) on Mozilla wiki.